library/Mockery/Reflector.php

```diff
@@ -205,14 +189,78 @@ private static function getTypeInformation(\ReflectionType $type, \ReflectionCla
      */
     private static function formatNullableType($typeHint)
     {
-        if (\PHP_VERSION_ID < 80000) {
-            return sprintf('?%s', $typeHint);
+        if ($typeHint === 'mixed') {
+            return $typeHint;
         }
 
-        if ($typeHint === 'null' || $typeHint === 'mixed') {
+        if (strpos($typeHint, 'null') !== false) {
             return $typeHint;
         }
 
+        if (\PHP_VERSION_ID < 80000) {
+            return sprintf('?%s', $typeHint);
+        }
+
         return sprintf('%s|null', $typeHint);
     }
+
+    public static function getTypeFromReflectionType(\ReflectionType $type, \ReflectionClass $declaringClass): string
+    {
+        if ($type instanceof \ReflectionNamedType) {
+            $typeHint = $type->getName();
+
+            if ($type->isBuiltin()) {
+                return $typeHint;
+            }
+
+            if ($typeHint === 'static') {
+                return $typeHint;
+            }
+
+            // 'self' needs to be resolved to the name of the declaring class
+            if ($typeHint === 'self'){
+                $typeHint = $declaringClass->getName();
+            }
+
+            // 'parent' needs to be resolved to the name of the parent class
+            if ($typeHint === 'parent'){
+                $typeHint = $declaringClass->getParentClass()->getName();
+            }
+
+            // class names need prefixing with a slash
+            return sprintf('\\%s', $typeHint);
+        }
+
+        if ($type instanceof \ReflectionIntersectionType) {
+            $types = array_map(
+                fn (\ReflectionType $type) => self::getTypeFromReflectionType($type, $declaringClass),
+                $type->getTypes()
+            );
+
+            return implode(
+                '&',
+                array_map(
+                    fn (string $type): string => $type,
+                    $types
+                )
+            );
```

Comment:
```suggestion
            return implode(
                '&',
                $types,
            );
```

II was going to suggest this, but maybe you wrapped it in array_map() because you want to guarantee in Runtime that $types is a list<string>?

Personally, I think @var list<string> $types type hinting would be fine.